### DISCUSSION TITLE
Message alerts instead of dialog boxes

### DISCUSSION THREAD

**Discussion by BazzaCuda** [Keriew/augustus#574] [Ideas]
Hi all,

In my own build, instead of getting a popup message dialog box that I have to dismiss, I just get a notification alert at the top of the window and the message goes into the messages list for me to read at my leisure (see screenshot - enclosed, I hope!).

I prefer not to be interrupted every five minutes with popup messages about stormy seas and collapsed mines etc. 

In the case of Emperor Requests, the neat thing is that with the Augustus sidebar info showing what's being requested and providing an action button, I can immediately dispatch the goods without first having to dismiss the usual dialog box/window.

Is this something that other people would be interested in having or that you think might be a useful tweak?
I would need to add an option in Settings to turn this on and off as preferred.

All comments, suggestions and opinions gratefully recieved - I'm happy to keep this one to myself if there are no takers :D

Baz.

![Clipboard Image (2)](https://user-images.githubusercontent.com/22550919/155206599-3a9d0990-4b83-4c4d-89aa-4d3c833d227b.jpg)

**Comment by crudelios** [comment]
I like this. Can you submit a PR, please? 

Pharaoh had something like this where you could select which messages had popups and which didn't.

**Reply by BazzaCuda** [reply]
I will do. The changes are currently in my development repository, not in my Augustus fork, so I need to reimplement it and test it all over again. Also, I haven't yet coded the Settings option for it. It'll take me a few days. I'll include my hotkey option for displaying the messages list, shall I?

**Reply by BazzaCuda** [reply]
PR submitted

**Comment by Ouaz** [comment]
Very good idea. Maybe additionally implement a hotkey to open directly the Messages Panel (without having to click in the sidebar icon), for people who wants to read the full message (new players for example).

**Reply by BazzaCuda** [reply]
I've got that too in my build :D

**Comment by Riww** [comment]
Played a game with this on, the problem is you are missing a lot. Scripted events should interrupt the game with popup: invasions/local uprisings, emperor requests, trade/price change events, gladiator revolt, earthquake, empire expansion - these all should popup. Un-interrupting messages are fine for the rest. Could you make an option for this? I don't think options for all possible messages are needed, just the ability to popup scripted events, and nothing else.

**Reply by BazzaCuda** [reply]
Admittedly, everyone's playing experience will differ, but I've been playing for weeks with this setting on and don't have any issues with it at all. I won't be returning to playing with the setting turned off. 

Some _friendly_ observations:
1. The setting is aimed at experienced users who are very familiar with the entire range of possible messages and which ones they need to deal with immediately and which ones can be deferred until a few moments later.
2. In the case of the former, the need for a hotkey to quickly show the message list, thereby acting as an interrupting popup, wrote itself.
3. In the case of e.g. invasions, for which most people immediately pause the game, I actually find it easier to pause without having to first dismiss the dialog box. The message alert and the invasion music is all that's required to prompt me to pause the game, even in the event of Barbarian Uprisings for which I haven't received over 3 years game time of warnings from Distant Battle to invasion.
4. Likewise with emperor requests, the info and action button is immediately displayed in the sidebar. The popup message is entirely redundant. I have yet to have any event which was so monumental that not having a message popup (rather than it going to the message list with an alert) made any difference, whatsoever. 
5. The change, as implemented, required 3 lines of code. Implementing categories of messages for which some should popup and others not is a significantly bigger modification. I suspect that, like me, when people get used to responding slightly differently to message alerts, they are likely to turn off all the categories of message popups anyway, making all that work redundant. Obviously, if someone wants to go to all that effort, that's entirely their choice. And if the dev team think that is a valuable use of their time to bring Augustus functionality in line with the similar settings that I hear are in Pharoah, that is entirely a matter for them. But I don't need it and neither do any of the several experienced users who road-tested the change for me....and now have message popups turned off permanently.
6. I think you are judging the setting harshly on the basis of having only played one game with it on. I would suggest this is not a reasonable user test. To suggest a major amount of work on that basis is unreasonable. The solution to your first statement is to simply ignore the setting entirely, as other people do with some of the other settings.
7. I  will concede though, that if you are whipping through the game at, say, 200% or more, to test the stability of your city, then the Pharoah-style categories _might_ be useful. But that scenario is a whole other discussion, I think. I've coped with invasions at 500% with no problems.

Please forgive me if any of my responses seem curt. They're not intended to be - I'm just being "to the point".  I used to be a [software] Product Manager for a large (well-known) software company and, sometimes, old habits die hard :D :D :D

Baz.

**Reply by Riww** [reply]
Currently, for me it is preferable to play with the option on, as the default interruption is simply too often. But I still like to be interrupted by popup with scripted events (invasions, emperor requests, and major events like earthquakes, revolts, empire expansions and Rome raises wages at least) . Yes, you can immediately pause the game and deal with the event in question, the problem is I often miss the message and by the time I get to it enemy soldiers are already destroying buildings in the city, all my soldiers still in forts. 
I believe a setting just for scripted events is not too much work.

**Reply by crudelios** [reply]
My own reply is at the bottom because I'm stupid and can't use these discussions properly (I feel like a boomer now).

Anyway, coding-wise, we could meet in the middle. Instead of having a setting for every popup like I mentioned below (which does seem like overkill), we could just have an option of "Default", "Limited" or "None". The code would still be relatively easy to implement and it would cater to most playstyles.

**Comment by crudelios** [comment]
We could add a new tab for the options window for this, then it's easy to add a setting for each message type. Even something like selecting "All/ default/minimal/none" could be done.
